src/tarantool-client/Converters/EnumConverter.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Globalization;
+
+using TarantoolDnx.MsgPack;
+
+namespace tarantool_client.Converters
+{
+    public class EnumConverter<T> : IMsgPackConverter<T>
+        where T : struct, IConvertible
+    {
+        public void Write(T value, IBytesWriter writer, MsgPackContext context)
```

Comment:
First of all, you should statically check that type is enum.
Second, not all enums based on int, they can be based on ulong,  for example.
Third, I prefer serialization to names.